### PR TITLE
add https request

### PR DESCRIPTION
**Pull request by ajamesa**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

https request

## Brief changelog

change http client

## Verifying this change

deploy nacos with https

Follow this checklist to help us incorporate your contribution quickly and easily:

* [x] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: client/src/main/java/com/alibaba/nacos/client/naming/net/HttpClient.java [hunk 1]
```diff
@@ -16,6 +16,7 @@
 package com.alibaba.nacos.client.naming.net;
 
 import com.alibaba.nacos.api.common.Constants;
+import com.alibaba.nacos.client.config.common.HttpsManager;
```
Comment (nkorange) [inline client/src/main/java/com/alibaba/nacos/client/naming/net/HttpClient.java]: naming module shouldn't have a dependency of config module.

Comment (ajamesa) [reply client/src/main/java/com/alibaba/nacos/client/naming/net/HttpClient.java]: naing and config package are both in the same moudule client

Comment (ajamesa) [reply client/src/main/java/com/alibaba/nacos/client/naming/net/HttpClient.java]: > naming module shouldn't have a dependency of config module.

naing and config package are both in the same moudule client

Comment (nkorange) [reply client/src/main/java/com/alibaba/nacos/client/naming/net/HttpClient.java]: > > naming module shouldn't have a dependency of config module.
> 
> naing and config package are both in the same moudule client

I mean logically, not physically. Naming is a submodule for service discovery, while Config is another submodule for configuration management.

Comment (ajamesa) [reply client/src/main/java/com/alibaba/nacos/client/naming/net/HttpClient.java]: > > > naming module shouldn't have a dependency of config module.
> > 
> > 
> > naing and config package are both in the same moudule client
> 
> I mean logically, not physically. Naming is a submodule for service discovery, while Config is another submodule for configuration management.

move them to cmmon pkg

Comment (nkorange) [reply client/src/main/java/com/alibaba/nacos/client/naming/net/HttpClient.java]: > > > > naming module shouldn't have a dependency of config module.
> > > 
> > > 
> > > naing and config package are both in the same moudule client
> > 
> > 
> > I mean logically, not physically. Naming is a submodule for service discovery, while Config is another submodule for configuration management.
> 
> move them to cmmon pkg

They should stay in client module, but in another package, may be: ``` com.alibaba.nacos.client.common```

Comment (ajamesa) [reply client/src/main/java/com/alibaba/nacos/client/naming/net/HttpClient.java]: > > > > > naming module shouldn't have a dependency of config module.
> > > > 
> > > > 
> > > > naing and config package are both in the same moudule client
> > > 
> > > 
> > > I mean logically, not physically. Naming is a submodule for service discovery, while Config is another submodule for configuration management.
> > 
> > 
> > move them to cmmon pkg
> 
> They should stay in client module, and in another package, may be: ` com.alibaba.nacos.client.common`

okay

File: client/src/main/java/com/alibaba/nacos/client/common/HttpsManager.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            sslContext.init(null, trustAllCerts, null);
+            HttpsURLConnection.setDefaultSSLSocketFactory(sslContext.getSocketFactory());
+        } catch (NoSuchAlgorithmException e) {
+            LOGGER.error("no such algorithn ", e);
+        } catch (KeyManagementException e) {
+            LOGGER.error("key management error ", e);
+        }
+    }
+
+    public static void verifieHttpsHostName(String url) {
```
Comment (nkorange) [inline client/src/main/java/com/alibaba/nacos/client/common/HttpsManager.java]: verifie --> verify

File: client/src/main/java/com/alibaba/nacos/client/common/SslConfig.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+/**
+ * @author James Chen
+ */
+public class SslConfig implements TrustManager, X509TrustManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(HttpsManager.class);
+    
+    @Override
+    public void checkClientTrusted(X509Certificate[] x509Certificates, String s) throws CertificateException {
+        LOGGER.info("ckeck client trust list {}", Arrays.asList(x509Certificates));
```
[the remark this answers is not in the document]
Comment (ajamesa) [reply client/src/main/java/com/alibaba/nacos/client/common/SslConfig.java]: Ok, delete it

[the remark this answers is not in the document]
Comment (ajamesa) [reply client/src/main/java/com/alibaba/nacos/client/common/SslConfig.java]: > Why in two log statements? and ckeck --> check

Ok, delete it

File: client/src/main/java/com/alibaba/nacos/client/naming/net/HttpClient.java [hunk 4]
```diff
@@ -104,7 +109,9 @@ public static HttpResult request(String url, List<String> headers, Map<String, S
 
             return new HttpResult(500, e.toString(), Collections.<String, String>emptyMap());
         } finally {
-            IoUtils.closeQuietly(conn);
+            if (conn != null) {
```
Comment (nkorange) [inline client/src/main/java/com/alibaba/nacos/client/naming/net/HttpClient.java]: Please user IoUtils.closeQuietly to keep connection reused. Refer to https://stackoverflow.com/questions/4767553/safe-use-of-httpurlconnection

**Comment by KomachiSion** [comment]
No response and update code for a long time, I will close this PR, if wanted to solve problem, please resubmit PR, Thanks.

**Pull request closed without merging**